docs/extending_deptrac.md

```diff
@@ -0,0 +1,54 @@
+# Extending Deptrac
+
+There are 2 main ways you can extend Deptrac:
+
+- [Output formatters](#Output Formatters) - To change how the output of dependency analysis is formatted. This might be useful to integrate Deptrac into your CI/CD pipeline.
+- [Commands](#Commands) - Take advantage of the Deptrac framework to analyse different aspects of your dependency tree.
```

Comment:
I have found commands to be extremely useful extension points thus far. If you look at deptrac-awesome it is basically a Better Graphviz formatter and the rest are commands to get some useful information from the resolved dependencies.
For the extension point you have mentioned:
 - InputCollectorInterface - I do not see any reasons why you might want to do something other than what config allows right now.
 - DependencyEmitterInterface - This is useful, however, any additions here would be more than likely additions to core deptrac itself (like a new AST node that can play a role in dependency) rather than a custom extension
 - LayerResolverInterface, CollectorInterface, This I wanted to mainly cover in the "other extension points" section.